### DISCUSSION TITLE
Dogfooding for huxleyhu

### DISCUSSION THREAD

**Discussion by husimplicity** [alibaba/GraphScope#2652] [dogfooding]
## Task 1: 在本地试用 GraphScope
- [x] 安装 GraphScope
- [x] 载入 `ogbn_mag` 内置图
- [x] Project `ogbn_mag`到一个简单图，选择点标签为 `paper`, 边标签为 `cites`, 属性都为空。
- [x] 在简单图上运行一个 Kcore 算法, 并打印结果为 pandas.DataFrame
- [x] 在 `ogbn_mag`图上启动交互引擎 （Interactive）
- [x] 通过 gremlin 计算图有多少点，边
- [x] 对于 `ogbn_mag`图，使用 learning 定义并执行一个 GCN 训练过程，可参考 CI 或文档中定义的过程
- [x] 退出 GraphScope。

## Task 2: 在本地开发调试 GraphScope

- [x] 通过 `vineyardd`直接启动 `vineyard`
- [x] 从源代码编译 Analytical Engine，找到编译出的binary `grape_engine`， 通过 `grape_engine`直接运行 analytical_engine，然后退出 grape_engine 进程
- [x] 为 GAE 添加一条日志 ( `analytical_engine/core/server/analytical_server.cc`)，并重新编译，启动并找到新添加的日志 
- [ ] 从源代码编译 Interactive Engine，找到编译出的 `frontend-x-SNAPSHOT.jar`和 `gaia_executor`,直接运行这个 jar 包和这个 gaia_executor，使 frontend 能够连接到 executor。
- [ ] 了解 Frontend 和 Executor 的代码位置和组织结构，为 GIE 的 Frontend (`interactive_engine/frontend/src/main/java/com/alibaba/graphscope/frontend/Frontend.java`)  和 executor (`interactive_engine/executor/assembly/v6d/src/bin/gaia_executor.rs`) 各添加一条日志，并重新编译，启动并找到新添加的日志 
- [x] 从源代码编译 Learning Engine, 找到 `libgraphlearn_shared.so`
- [x] 使用 setup.py 安装 coordinator 和 client。 注意因依赖关系， client 安装要在 coordinator 之前
- [ ] 继续安装 GAE，GIE，和 GLE 到预设安装目录。（上一步只有编译。相当于 make 和 make install 的区别）
- [ ] 找到 Client， Coordinator， GAE，GIE 和 GLE 的安装位置目录
- [ ] 若一切正常，这时你能通过 python 拉起 graphscope。启动python，import graphscope 并载入一个内置 modern_graph 图，在图上启动图交互引擎，发送一个查所有点的语句。
- [x] 退出 graphscope

## System Info
```
$ uname -a
Linux iZt4n5ydg0eiyxpvlo34vuZ 5.4.0-137-generic #154-Ubuntu SMP Thu Jan 5 17:03:22 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux

$ python3 --version
Python 3.10.6
```

**Comment by husimplicity** [comment]
1. java command not found => install jdk
2. default tensorflow==2.12 conflicts with dependency of graphscope, fixed by install tensorflow==2.8

**Reply by sighingnow** [reply]
Fixed by #2650

**Comment by husimplicity** [comment]
```
[graphscope@j63c09243 GraphScope]$ sudo make interactive
error: linking with `cc` failed: exit status: 1
error: could not compile `gaia_runtime` due to previous error; 1 warning emitted
[ERROR] Failed to execute goal org.codehaus.mojo:exec-maven-plugin:3.0.0:exec (compile-exec-id) on project executor: Command execution failed.: Process exited with an error: 101 (Exit value: 101) -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <args> -rf :executor
make: *** [/home/graphscope/GraphScope//interactive_engine/assembly/target/graphscope.tar.gz] Error 1


环境：dev container

**Reply by sighingnow** [reply]
@longbinlai

**Reply by husimplicity** [reply]
[graphscope@j63c09243 GraphScope]$ ./gs make interactive --storage-type=vineyard
Setting the environment for development.

[WARNING] GRAPHSCOPE_HOME will set to source root (/home/graphscope/GraphScope) for development.
[WARNING] To use you assigned GRAPHSCOPE_HOME, export GRAPHSCOPE_ENV=prod.

Read the env: GRAPHSCOPE_HOME=/home/graphscope/GraphScope
invalid option: --storage-type
